File: app/src/main/java/com/zia/gankcqupt_mvp/Adapter/PagerAdapter.java
package com.zia.gankcqupt_mvp.Adapter;

import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;

import com.zia.gankcqupt_mvp.View.Fragment.Page.MeFragment;
import com.zia.gankcqupt_mvp.View.Fragment.Page.SearchFragment;
import com.zia.gankcqupt_mvp.View.Fragment.Page.SortFragment;

/**
 * Created by zia on 2017/5/18.
 */

public class PagerAdapter extends FragmentPagerAdapter {

    private String[] mTitles = new String[]{"查询", "分类", "更多"};

    public PagerAdapter(FragmentManager fm) {
        super(fm);
    }

    @Override
    public Fragment getItem(int position) {
        if(position == 0){
            return new SearchFragment();
        }else if(position == 1){
            return new SortFragment();
        }else if(position == 2){
            return new MeFragment();
        }
        return new Fragment();
    }

    @Override
    public int getCount() {
        return mTitles.length;
    }

    @Override
    public CharSequence getPageTitle(int position) {
        return mTitles[position];
    }
}

File: app/src/main/java/com/zia/gankcqupt_mvp/ui/empty/PagerAdapter.kt
package com.zia.gankcqupt_mvp.ui.empty

import android.support.v4.app.Fragment
import android.support.v4.app.FragmentManager
import android.support.v4.app.FragmentPagerAdapter
import java.util.*

/**
 * Created by zia on 2018/3/7.
 */
class PagerAdapter(fm: FragmentManager) : FragmentPagerAdapter(fm) {

    private val fragments = ArrayList<ContentFragment>()
    private val count = 5

    fun freshData(empty: String) {
        try {
            val rooms = empty.substring(1, empty.length - 1)
                    .replace(" ", "")
                    .split(",")
            val trans = arrayListOf("2", "3", "4", "5", "8")
            for (i in 0 until trans.size) {
                var old = ""
                val sb = StringBuilder("")
                rooms.forEach {
                    if (old != it.substring(1, 2)) {
                        old = it.substring(1, 2)
                        sb.append("\n\n\n")
                    }
                    if (it.substring(0, 1) == trans[i])
                        sb.append(it).append("    ")
                }
                fragments[i].setText(sb.toString().trim())
            }
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }

    init {
        for (i in 0 until count) {
            fragments.add(ContentFragment())
        }
    }

    override fun getItem(position: Int): Fragment {
        return fragments[position]
    }

    override fun getCount(): Int {
        return fragments.size
    }

    private val trans = arrayListOf("二教", "三教", "四教", "五教", "八教")
    override fun getPageTitle(position: Int): CharSequence? {
        return trans[position]
    }
}
